src/PantheonTerminalWindow.vala

```diff
@@ -305,6 +316,21 @@ namespace PantheonTerminal {
             zoom_default_button.clicked.connect (() => action_zoom_default_font ());
             zoom_out_button.clicked.connect (() => action_zoom_out_font ());
 
+            color_button.mode_changed.connect (() => {
```

Comment:
What should happen here:

The `mode_changed` signal should set `settings.prefer_dark_style`.

Then you connect to `settings.notify["prefer-dark-style"]` and change the background and foreground based on that value. (You make an `update_style` or something).

Reply:
In the case that we add a third style (for example a high contrast black on white style), prefering the dark stylesheet does not necessarily imply what the background and foreground colors should be.